src/main/java/com/example/onlinebookshop/services/impl/OrderServiceImpl.java

```diff
@@ -0,0 +1,175 @@
+package com.example.onlinebookshop.services.impl;
+
+import com.example.onlinebookshop.dto.order.request.CreateOrderDto;
+import com.example.onlinebookshop.dto.order.request.UpdateOrderDto;
+import com.example.onlinebookshop.dto.order.response.OrderDto;
+import com.example.onlinebookshop.dto.orderitem.response.OrderItemDto;
+import com.example.onlinebookshop.entities.Order;
+import com.example.onlinebookshop.entities.Order.Status;
+import com.example.onlinebookshop.entities.OrderItem;
+import com.example.onlinebookshop.entities.ShoppingCart;
+import com.example.onlinebookshop.exceptions.AddressNotFoundException;
+import com.example.onlinebookshop.exceptions.EntityNotFoundException;
+import com.example.onlinebookshop.exceptions.OrderProcessingException;
+import com.example.onlinebookshop.mapper.OrderItemMapper;
+import com.example.onlinebookshop.mapper.OrderMapper;
+import com.example.onlinebookshop.repositories.order.OrderRepository;
+import com.example.onlinebookshop.repositories.orderitem.OrderItemRepository;
+import com.example.onlinebookshop.repositories.shoppingcart.ShoppingCartRepository;
+import com.example.onlinebookshop.services.OrderService;
+import jakarta.transaction.Transactional;
+import java.math.BigDecimal;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.List;
+import java.util.Optional;
+import lombok.RequiredArgsConstructor;
+import org.springframework.data.domain.Pageable;
+import org.springframework.stereotype.Service;
+
+@Service
+@RequiredArgsConstructor
+public class OrderServiceImpl implements OrderService {
+    private final DateTimeFormatter formatter =
+            DateTimeFormatter.ofPattern("yyyy-MM-dd'T'HH:mm:ss");
+    private final OrderRepository orderRepository;
+    private final ShoppingCartRepository shoppingCartRepository;
+    private final OrderItemRepository orderItemRepository;
+    private final OrderMapper orderMapper;
+    private final OrderItemMapper orderItemMapper;
+
+    @Override
+    public List<OrderDto> getOrders(Pageable pageable) {
+        return orderRepository.findAll().stream()
+                .map(orderMapper::orderToOrderDto)
+                .toList();
+    }
+
+    @Override
+    public List<OrderDto> getOrdersByUserId(Long userId, Pageable pageable) {
+        return orderRepository.findAllByUserId(userId, pageable).stream()
+                .map(orderMapper::orderToOrderDto)
+                .toList();
+    }
+
+    @Transactional
+    @Override
+    public OrderDto addOrder(Long userId, CreateOrderDto createOrderDto) {
+        if (createOrderDto.shippingAddress().isBlank()) {
```

Comment:
You have NotBlank validation, so I think this check is redundant. 
Also, remove the exception